qiskit/circuit/add_control.py

```diff
@@ -93,6 +91,8 @@ def control(
     # pylint: disable=cyclic-import
     from qiskit.circuit import controlledgate
 
+    ctrl_state = _ctrl_state_to_int(ctrl_state, num_ctrl_qubits)
```

Comment:
Could you add a test to cover this case with a string and None as input (like the examples you used to show it breaks), plus a `fixes` entry in the release note to explain this additional bugfix? 🙂